.github/workflows/kfp-backend-postgres-tests.yml

```diff
@@ -0,0 +1,245 @@
+          fi
+          echo "local-path-provisioner namespace: $LPP_NS"
+
+          # 3) Wait for controller to be ready
+          kubectl -n "$LPP_NS" rollout status deploy/local-path-provisioner --timeout=180s
+
+          # 4) Mark StorageClass as default (idempotent)
+          if ! kubectl get storageclass local-path >/dev/null 2>&1; then
+            echo "StorageClass 'local-path' not found, listing SCs:"
+            kubectl get storageclass -o wide
+            exit 1
+          fi
+          kubectl annotate storageclass local-path storageclass.kubernetes.io/is-default-class="true" --overwrite
+
+          echo "StorageClasses:"
+          kubectl get storageclass
+
+          # 5) Verify default StorageClass exists
+          kubectl get storageclass | tee /dev/stderr | grep "(default)"
+
+      - name: Wait for default StorageClass present
+        run: |
+          # Sanity check: ensure there is a default StorageClass before deploying PG
+          kubectl get storageclass | tee /dev/stderr | grep "(default)" >/dev/null
+
+      - name: Deploy Postgres via Kustomize
+        working-directory: manifests/kustomize/third-party/postgresql
+        env:
+          NAMESPACE: kfp-pgx-test
+        run: |
+          kustomize build ./base | kubectl -n "$NAMESPACE" apply -f -
+          kubectl -n "$NAMESPACE" get all
+
+      - name: Wait for Postgres pod ready
+        env:
+          NAMESPACE: kfp-pgx-test
+        run: |
+          # Wait until pods with label=app=postgres are Ready
+          kubectl -n "$NAMESPACE" wait --for=condition=ready pod -l app=postgres --timeout=300s
+          kubectl -n "$NAMESPACE" get pods -l app=postgres -o wide
+
+      - name: Port-forward Postgres (bind to 127.0.0.3)
+        env:
+          NAMESPACE: kfp-pgx-test
+        run: |
+          set -euo pipefail
+
+          kubectl -n "$NAMESPACE" get svc postgres-service -o wide
+          SVC_PORT=$(kubectl -n "$NAMESPACE" get svc postgres-service -o jsonpath='{.spec.ports[0].port}')
+
+          kubectl -n "$NAMESPACE" port-forward svc/postgres-service ${SVC_PORT}:${SVC_PORT} --address=127.0.0.3 >/tmp/pf.log 2>&1 &
+          echo $! > pf.pid
+
+          # Wait until endpoints are ready
+          for i in {1..150}; do
+            EP=$(kubectl -n "$NAMESPACE" get endpoints postgres-service -o jsonpath='{.subsets[*].addresses[*].ip}' 2>/dev/null || true)
+            [ -n "$EP" ] && break
+            sleep 2
+          done
+          [ -n "${EP:-}" ] || { echo "no endpoints"; cat /tmp/pf.log || true; exit 1; }
+
+          # Wait until local port is open
+          for i in {1..60}; do nc -z 127.0.0.3 "${SVC_PORT}" && break; sleep 2; done
+          nc -z 127.0.0.3 "${SVC_PORT}" || { echo "local port not open"; cat /tmp/pf.log || true; exit 1; }
+
+          # Optional: psql probe (not fatal; default db/user may not exist)
+          PGPASSWORD=kubeflow psql -h 127.0.0.3 -p "${SVC_PORT}" -U kubeflow -d kubeflow -c "SELECT 1;" || true
```

Comment:
I don't think we need these, the postgresql deployment already has a readiness probe, so if the pod is ready, the svc should be ready. And for the second one, we can assume kubectl port-forward is working as expected.